packages/vite/src/node/config.ts

```diff
@@ -270,6 +287,11 @@ export async function resolveConfig(
   // user config may provide an alternative mode
   mode = config.mode || mode
 
+  // user config may provide alternative default known js src extensions
+  if (!config.knownJsSrcExtensions) {
+    config.knownJsSrcExtensions = [...DEFAULT_KNOWN_JS_SRC_EXTENSIONS]
+  }
```

Comment:
We can wait until we discuss this with Evan, but I think it is better if this works like [`assetsInclude`](https://vitejs.dev/config/#assetsinclude). So the user will give us extra extensions and they are added to the ones supported by default by vite.